src/main/java/org/elasql/remote/groupcomm/server/ConnectionMgr.java

```diff
@@ -93,6 +96,8 @@ public void onServerFailed(int failedServerId) {
 
 	@Override
 	public void onReceiveP2pMessage(ProcessType senderType, int senderId, Serializable message) {
+		// For controller
+		TransactionProfiler.getLocalProfiler().incrementNetworkInSize(message);

```

Comment:
This line should be put at `TPartCacheMgr` because the worker thread does not come here.